From dd75cf7405b7fa8206b03113fb49aec12454e5bd Mon Sep 17 00:00:00 2001
From: Jakub Wilk <jwilk@jwilk.net>
Date: Wed, 26 Aug 2020 10:58:53 +0200
Subject: [PATCH] debmany: use apt-helper for package downloads

apt-helper is better that curl, because:
* APT it's more likely to be already installed (popcon 100%) than curl
  (popcon 55%).
* curl doesn't support all the URL schemes APT does.
* curl doesn't honor APT proxy settings.
---
 debmany/debmany          | 8 ++++----
 debmany/man/debmany.1    | 2 +-
 debmany/man/debmany.de.1 | 2 +-
 3 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/debmany/debmany b/debmany/debmany
index 3542d8b..6c08197 100755
--- a/debmany/debmany
+++ b/debmany/debmany
@@ -4,7 +4,7 @@
 # description: select manpages+docs of any [not] installed Debian package
 # comment:     debmany is based on the work of debman and debget
 # gui uses:    whiptail (preferred), dialog or zenity as well
-# other uses:  curl, $PAGER or sensible-pager (can be overwritten: -o)
+# other uses:  apt, $PAGER or sensible-pager (can be overwritten: -o)
 # optional:    -k -x: kfmclient exec (konqueror), xdg-open (xdg-utils)
 # This is free software.  You may redistribute copies of it under the terms of
 # the GNU General Public License <https://www.gnu.org/licenses/gpl.html>.
@@ -350,9 +350,9 @@ else
         fi
       fi
       # download the package
-      debug "Mode3: Start download using curl" # comment
-      curl --location "$url" >"$file" || error "Failed to download '$url' to '$file'."
-      status=repos-curl
+      debug "Mode3: Start download using APT" # comment
+      /usr/lib/apt/apt-helper download-file "$url" "$file" || error "Failed to download '$url' to '$file'."
+      status=repos-download
     fi
   fi
   cd "$temp"
diff --git a/debmany/man/debmany.1 b/debmany/man/debmany.1
index cd0ad5f..6c29383 100644
--- a/debmany/man/debmany.1
+++ b/debmany/man/debmany.1
@@ -43,7 +43,7 @@ Zenity (a graphical dialog) is used to select the files and if necessary to ente
 .PP
 The manpages are temporarily extracted to a temporary directory under /tmp (or /var/tmp, if /tmp is not a directory). You can override the path if you set and export the environment variable TMPDIR or set it as a temporary variable.
 .PP
-You need to have the Debian package "whiptail" (preferred) or "dialog" as well as "curl" installed.
+You need to have the Debian package "whiptail" (preferred) or "dialog" as well as "apt" installed.
 .SH "EXAMPLES"
 .TP
 debmany coreutils
diff --git a/debmany/man/debmany.de.1 b/debmany/man/debmany.de.1
index 05cacf1..8b3009a 100644
--- a/debmany/man/debmany.de.1
+++ b/debmany/man/debmany.de.1
@@ -43,7 +43,7 @@ Zur Auswahl der Dateien und ggf. zur Eingabe des Paketes wird "zenity" (grafisch
 .PP
 Die Manpages werden temporär nach /tmp oder /var/tmp extrahiert. Durch das Setzen und Exportieren der Umgebungsvariable TMPDIR bzw. das Setzen als temporären Variable kann der Speicherort überschrieben werden.
 .PP
-Das Debian Paket "whiptail" (bevorzugt) oder "dialog" sowie "curl" muss installiert sein.
+Das Debian Paket "whiptail" (bevorzugt) oder "dialog" sowie "apt" muss installiert sein.
 .SH "BEISPIELE"
 .TP
 debmany coreutils
-- 
2.28.0

